sympy/solvers/ode/ode.py

```diff
@@ -1752,6 +1752,209 @@ def match_2nd_linear_bessel(r, func):
     rn['b4'] = point
     return rn
 
+
+# find coefficients of terms f(t), diff(f(t),t) and higher derivatives
+# and similarly for other functions g(t), diff(g(t),t) in all equations.
+# Here j denotes the equation number, funcs[l] denotes the function about
+# which we are talking about and k denotes the order of function funcs[l]
+# whose coefficient we are calculating.
+def linearity_check(eqs, j, func, funcs, func_coef, order, t, is_linear_):
```

Comment:
Since we're not using this function in the new code any more it would be better to put it back where it was and leave the old code unchanged.

Reply:
Ok but I will keep the ```_get_funcs_order``` function outside as it can be helpful for getting the order of the differential terms in the equations.